crutest/src/main.rs

```diff
@@ -2045,24 +2047,32 @@ async fn fill_sparse_workload(
 ) -> Result<()> {
     let mut rng = rand_chacha::ChaCha8Rng::from_entropy();
 
-    // Figure out how many extents we have
-    let extents = ri.total_blocks / (ri.extent_size.value as usize);
-    let extent_size = ri.extent_size.value as usize;
-
-    // Do one write to each extent.
-    for extent in 0..extents {
-        let mut block_index: usize = extent * extent_size;
-        let random_offset: usize = rng.gen_range(0..extent_size);
-        block_index += random_offset;
+    // We loop over all sub volumes, doing one write to each extent.
+    for (index, sv) in ri.volume_info.volumes.iter().enumerate() {
+        let extents = sv.extent_count as usize;
+        let extent_size = sv.blocks_per_extent as usize;
+        for extent in 0..extents {
+            let mut block_index: usize = extent * extent_size;
+            let random_offset = rng.gen_range(0..extent_size);
+            block_index += random_offset;
 
-        let offset = BlockIndex(block_index as u64);
+            let offset = BlockIndex(block_index.try_into().unwrap());
 
-        ri.write_log.update_wc(block_index);
+            ri.write_log.update_wc(block_index);
 
-        let data = fill_vec(block_index, 1, &ri.write_log, ri.block_size);
+            let data = fill_vec(
+                block_index,
+                1,
+                &ri.write_log,
+                ri.volume_info.block_size,
+            );
 
-        println!("[{extent}/{extents}] Write to block {}", block_index);
-        volume.write(offset, data).await?;
+            println!(
+                "[{index}][{extent}/{extents}] Write to block {}",
+                block_index
+            );
+            volume.write(offset, data).await?;
```

Comment:
This doesn't look right to me, because `offset` here is a relative offset within the subvolume (but ignores the offset of that subvolume within the whole volume).

I think we need to accumulate a global offset (cumulative sum of `sv.extent_count * sv.blocks_per_extent`) and add it to `offset` here.

Reply:
Yes, this was wrong and not doing what I wanted.
I've updated it with an additional global offset and I can see from the test output that it is touching all extents